ddtrace/contrib/pytest/plugin.py

```diff
@@ -116,29 +192,48 @@ def pytest_runtest_protocol(item, nextitem):
         yield
         return
 
-    with _CIVisibility._instance.tracer.trace(
+    if isinstance(item.parent.parent, pytest.Package):
+        if _extract_span(item.parent.parent) is None:
+            _start_test_module_span(item)
+
+    if _extract_span(item.parent) is None:
+        _start_test_suite_span(item)
+    module_span = _extract_span(item.parent)
+
+    with _CIVisibility._instance.tracer.start_span(
         ddtrace.config.pytest.operation_name,
+        child_of=module_span,
         service=_CIVisibility._instance._service,
         resource=item.nodeid,
         span_type=SpanTypes.TEST,
+        activate=True,
     ) as span:
         span.set_tag_str(COMPONENT, "pytest")
-        span.set_tag(SPAN_KIND, KIND)
-        span.set_tag(test.FRAMEWORK, FRAMEWORK)
-        span.set_tag(test.NAME, item.name)
+        span.set_tag_str(SPAN_KIND, KIND)
+        span.set_tag_str(test.FRAMEWORK, FRAMEWORK)
+        span.set_tag_str(_EVENT_TYPE, SpanTypes.TEST)
+        span.set_tag_str(test.NAME, item.name)
+        span.set_tag(_SESSION_ID, span.trace_id)
+        test_suite_span = _extract_span(item.parent)
+        span.set_tag(_SUITE_ID, test_suite_span.span_id)
+
+        if isinstance(item.parent.parent, pytest.Package):
```

Comment:
are there cases where there'll be no module?

Reply:
Rarely, but yes. For example, if someone doesn't have a test module (no `__init__.py` file in the module where the test file is contained) and runs pytest on a test file directly, i.e. `pytest test_abc.py` there will be no packages recognized by pytest.